src/Table.cpp

```diff
@@ -346,4 +346,178 @@ namespace galsim {
 
     template class Table<double,double>;
 
+
+    // ArgGrid
+
+    template<class A>
+    ArgGrid<A>::ArgGrid(const A* args, int N)
+    {
+        grid.reserve(N);
+        for (int i=0; i<N; i++, args++) grid.push_back(*args);
+        const double tolerance = 0.01;
+        da = (grid.back() - grid.front()) / (N-1);
+        if (da == 0.) throw TableError("First and last arguments are equal.");
+        equalSpaced = true;
+        for (int i=1; i<N; i++) {
+            if (std::abs((grid[i] - grid.front())/da - i) > tolerance) equalSpaced = false;
+            if (grid[i] <= grid[i-1])
+                throw TableError("Table2D arguments not strictly increasing.");
+        }
+        lastIndex = 1;
+        lower_slop = (grid[1]-grid[0]) * 1.e-6;
+        upper_slop = (grid[N-1]-grid[N-2]) * 1.e-6;
+    }
+
+    // Look up an index.  Use STL binary search.
+    template<class A>
+    int ArgGrid<A>::upperIndex(const A a) const {
```

Comment:
I went ahead and reimplemented Table using ArgGrid (which I renamed ArgVec, since grid didn't feel right, especially in 1D).  In the process, I removed (well, declined to reimplement) a few items that didn't seem to be getting used, like constructing a Table from a stream.